### DISCUSSION TITLE
Livecharts requires python for compiling ?

### DISCUSSION THREAD

**Discussion by digger73** [unoplatform/uno#15038] [General]
I have installed livecharts 2.0.0-rc2 in my UNO project, when compiling to my surprise the WASM project comes up with an error regarding "Python is not installed. You can install....". Both mobile and windows projects builds without any requirements towards python.

Can it really be true, that the WASM project suddenly needs python to be able to compile due to livecharts ? (the other projects doesn't)
From my unknowing point of view it seems like an unnecessary dependency ?

**Comment by limeniye** [comment]
https://thenewstack.io/python-and-webassembly-elevating-performance-for-web-apps/
>By [using WebAssembly](https://thenewstack.io/webassembly/what-is-webassembly-and-why-do-you-need-it/), Python code can be compiled into highly optimized, low-level binary code that runs at near-native speed, significantly enhancing application performance and reducing network latency.

**Reply by digger73** [reply]
"Python code can be compiled", but I'm using UNO writing in C# ?

**Reply by jeromelaban** [reply]
@digger73 thanks for the question, yes, python is required because livecharts depends on SkiaSharp which is contains a native Wasm dependency. Python is used by emscripten, which creates the final webassembly binary.

**Reply by digger73** [reply]
@jeromelaban 
Thank you for the clarification, maybe add it to the documentation as I couldn't find that Python was a needed install. Just a shame as it adds further complexity to a CI/CD environment, which would be nice not to have.
I looked at the other 3'rd party control libraries, but none of them seems to give the same feel or are not actively developed up on (Infragistic are not actively working on it, syncfusion haven't made any changes for 3 years, lightningchart seems to be UWP). 

Are you aware of other components available for creating graphs and supports UNO.

**Reply by jeromelaban** [reply]
Python is requested by uno.check, and the build fails with a documentation link to explain what to do. As for the shame, that's what the underlying tooling for webassembly is required at this time.

For components, there's also https://github.com/HavenDV/H.OxyPlot. For others, if you need additional support from Telerik, for instance, make sure to comment and upvote [here](https://feedback.telerik.com/winui/1499061-add-support-for-uno-platform) or [here](https://feedback.telerik.com/uwp/1430657-add-support-for-uno-platform).

**Reply by digger73** [reply]
That is strange, I haven't had UNO.check to fail due to python ?

**Reply by limeniye** [reply]
uno.check, by default, has an Python installation check.
![image](https://github.com/unoplatform/uno/assets/69314237/4b2225d0-5682-4207-81fd-5fa558cc62d4)
Which version of uno-check are you using?
